cmd/nvidia-dra-plugin/nvlib.go

```diff
@@ -46,32 +47,39 @@ func newDeviceLib(driverRoot root) (*deviceLib, error) {
 		return nil, fmt.Errorf("failed to locate nvidia-smi: %w", err)
 	}
 
-	// In order for nvidia-smi to run, we need to set the PATH to the parent of
-	// the nvidia-smi executable and update LD_PRELOAD to include the path to
-	// libnvidia-ml.so.1
-	updatePathListEnvvar("LD_PRELOAD", driverLibraryPath)
-	updatePathListEnvvar("PATH", filepath.Dir(nvidiaSMIPath))
-
 	// We construct an NVML library specifying the path to libnvidia-ml.so.1
 	// explicitly so that we don't have to rely on the library path.
 	nvmllib := nvml.New(
 		nvml.WithLibraryPath(driverLibraryPath),
 	)
 	d := deviceLib{
-		Interface:     nvdev.New(nvdev.WithNvml(nvmllib)),
-		nvmllib:       nvmllib,
-		nvidiaSMIPath: nvidiaSMIPath,
+		Interface:         nvdev.New(nvdev.WithNvml(nvmllib)),
+		nvmllib:           nvmllib,
+		driverLibraryPath: driverLibraryPath,
+		nvidiaSMIPath:     nvidiaSMIPath,
 	}
 	return &d, nil
 }
 
-// updatePathListEnvvar prepends a specified list of strings to a specified envvar.
-func updatePathListEnvvar(envvar string, prepend ...string) {
+// prependPathListEnvvar prepends a specified list of strings to a specified envvar and returns its value.
+func prependPathListEnvvar(envvar string, prepend ...string) string {
 	if len(prepend) == 0 {
-		return
+		return os.Getenv(envvar)
 	}
 	current := filepath.SplitList(os.Getenv(envvar))
-	os.Setenv(envvar, strings.Join(append(prepend, current...), string(filepath.ListSeparator)))
+	return strings.Join(append(prepend, current...), string(filepath.ListSeparator))
+}
+
+// addOrUpdateEnvvarList adds or updates an envar to the list of specified envvars.
+func addOrUpdateEnvvarList(envvars *[]string, key, value string) {
+	for i, envvar := range *envvars {
+		pair := strings.SplitN(envvar, "=", 2)
+		if pair[0] == key {
+			(*envvars)[i] = fmt.Sprintf("%s=%s", key, value)
+			return
+		}
+	}
+	*envvars = append(*envvars, fmt.Sprintf("%s=%s", key, value))
```

Comment:
Is creating a new output not easier to follow?
```suggestion
func ensureEnvvar(envvars []string, key, value string) []string {
	var updated []string
	for i, envvar := range envvars {
		pair := strings.SplitN(envvar, "=", 2)
		if pair[0] == key {
			continue	         
		}
		updated = append(updated, envvar)
	}
	return = append(updated, fmt.Sprintf("%s=%s", key, value))
```

Note that since later occurrences of an evvar automatically take precendence, this logic may not be needed at all as we can simply apped the new pair.

Reply:
That's true, but it feels wrong to rely on this somehow, knowing that the envvar most definitely is already in the list (at least for the case of PATH).